File: LeetCode/JavaScript/Two Pointers/Easy/valid-palindrome.js
/**
 * Question Link: https://leetcode.com/problems/valid-palindrome/
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {

    str = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase(); // replace func will remove non-alphanumeric char
    let leftPtr = 0;
    let rightPtr = str.length - 1;

    while (leftPtr <= rightPtr) {
        if (str[leftPtr] !== str[rightPtr]) {
            return false;
        }
        leftPtr++;
        rightPtr--;
    }

    return true;
};

File: LeetCode/NeetCode/Python/Two Pointers/Easy/valid-palindrome.py
# Question Link: https://leetcode.com/problems/valid-palindrome/

import re

class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """

        new_str = re.sub(r'[^A-Za-z0-9]', '', s)
        new_str = new_str.lower()

        leftPtr = 0
        rightPtr = len(new_str) - 1

        while(leftPtr <= rightPtr):
            if(new_str[leftPtr] != new_str[rightPtr]):
                return False

            leftPtr += 1
            rightPtr -= 1
            

        return True
        